Unpeg numpy==1.23.4 by switching to igor2

Currently we have pegged `numpy==1.23.4` in our requirements.

This is because in `numpy>=1.23` the `np.complex` which was an alias for the builtin `complex` is deprecated.

We encounter this error courtesy of the `igor` library.

```bash
❱ pip show numpy | head -n20
Name: numpy
Version: 1.26.0
Summary: Fundamental package for array computing in Python
Home-page: https://numpy.org
Author: Travis E. Oliphant et al.
Author-email: 
License: Copyright (c) 2005-2023, NumPy Developers.
❱ pytest -n 14
ImportError while loading conftest '/home/neil/work/projects/topostats/TopoStats/tests/conftest.py'.
tests/conftest.py:14: in <module>
    from topostats.io import read_yaml, LoadScans
topostats/io.py:15: in <module>
    from igor import binarywave
/home/neil/.virtualenvs/topostats/lib/python3.11/site-packages/igor/binarywave.py:110: in <module>
    1:_numpy.complex, # NT_CMPLX, makes number complex.
/home/neil/.virtualenvs/topostats/lib/python3.11/site-packages/numpy/__init__.py:324: in __getattr__
    raise AttributeError(__former_attrs__[attr])
E   AttributeError: module 'numpy' has no attribute 'complex'.
E   `np.complex` was a deprecated alias for the builtin `complex`. To avoid this error in existing code, use `complex` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.complex128` here.
E   The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E       https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

A drop-in replacement that may solve this is to switch from [igor==0.3](https://pypi.org/project/igor/) which was last updated 2016-08-02 to [igor2==0.5.3](https://pypi.org/project/igor/) which was last updated 2023-07-21.

From [Igor2 README](https://github.com/AFM-analysis/igor2)...

> Igor2 is the continuation of the inactive igor project, forked from W. Trevor King and originally written by Paul Kienzle (see git history).

...and...

> This package is a direct replacement of igor. Your scripts should work without any issues if you replace:
> `import igor`
>
> with:
> 
> `import igor2 as igor`
>
> See the docstrings and unit tests for examples using the Python API.

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -12,7 +12,7 @@ jobs:
     strategy:
       matrix:
         os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11']
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -19,9 +19,9 @@ jobs:
         uses: actions/setup-python@v4.3.0
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install dependencies
+      - name: Upgrade pip and install test dependencies
         run: |
-          pip install -e .
+          pip install --upgrade pip
           pip install -e .[tests]
       - name: Test with pytest
         run: |

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,9 +37,9 @@ keywords = [
 ]
 requires-python = ">=3.8"
 dependencies = [
-  "igor",
+  "igor2",
   "matplotlib",
-  "numpy==1.23.4",
+  "numpy",
   "pandas",
   "pySPM",
   "pyyaml",
@@ -78,10 +78,8 @@ docs = [
   "sphinxcontrib-napoleon",
 ]
 dev = [
-  "Flake8-pyproject",
   "black",
-  "flake8",
-  "flake8-print",
+  "ipython",
   "pre-commit",
   "pylint",
   "pyupgrade",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,7 @@ keywords = [
 ]
 requires-python = ">=3.8"
 dependencies = [
-  "igor2",
+  "igor2>=0.5.3",
   "matplotlib",
   "numpy",
   "pandas",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -35,9 +35,9 @@ keywords = [
   "afm",
   "image processing"
 ]
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 dependencies = [
-  "igor2>=0.5.3",
+  "igor2",
   "matplotlib",
   "numpy",
   "pandas",

--- a/topostats/io.py
+++ b/topostats/io.py
@@ -12,7 +12,7 @@ from typing import Any
 import numpy as np
 import pandas as pd
 import pySPM
-from igor import binarywave
+from igor2 import binarywave
 import tifffile
 import h5py
 from ruamel.yaml import YAML, YAMLError

